### PR TITLE
stable/jenkins: option to set existing secret with Google Application Default Credentials

### PR DESCRIPTION
**Pull request by asmirnou**
#### What this PR does / why we need it:

The chart had the option for specifying the existing secret containing the AWS credentials. The underlying code created AWS-specific environmental variables for backup POD. Despite of the fact that backup utility is able to copy files to different cloud storages, this helm chart could not configure Google Cloud Storage. Google application credentials are kept in a file, which has to be mounted to a pod. This PR adds an option to set `gcpcredentials` in `existingSecret` as follows:

```
 existingSecret:
    jenkins-service-account:
      gcpcredentials: application_default_credentials.json
```

Helm template then creates the necessary volume mounts and `GOOGLE_APPLICATION_CREDENTIALS` envrionmental variable. So with this change, developer now can configure either AWS or GCP storage. 

It also adds and option to set backup pod labels, which are necessary to apply custom network policies. For example, if default policy denies everything, an application then can define custom rules to whitelist the connections it needs. So it needs labels for that.

AWS-specific defaults have been commented for backup module, as they have to be set explicitly.

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/jenkins]`)

### CODE REVIEW

**Comment by torstenwalter** [comment]
looks good as far as I can tell. @maorfr Can you have a look as you've been working on backup topics?

File: stable/jenkins/values.yaml [hunk 1]
```diff
@@ -504,10 +505,13 @@ backup:
   ## Use this key for AWS secret access key
      # awssecretkey: jenkins_aws_secret_key
   # Add additional environment variables
+   # jenkinsgcp:
+  ## Use this key for GCP credentials
+     # gcpcredentials: credentials.json
   env:
```
Comment (torstenwalter) [inline stable/jenkins/values.yaml]: when removing the values it might make sense to set the default to `[]`

**Comment by torstenwalter** [comment]
@asmirnou can you merge master into this branch?
The test failed as the chart version needs to be increased.
Could you also add an entry in CHANGELOG.md?

**Comment by asmirnou** [comment]
Sure, will do. Thanks!

**Comment by maorfr** [comment]
Looks very good to me!

@asmirnou, thanks for this!
If you're looking for more contributions, a similar change can be done for incubator/cassandra.

Very nice work.

**Comment by torstenwalter** [comment]
@asmirnou Looks great. Could you add an entry in CHANGELOG?

**Pull request merged**